### PR TITLE
Optimize the batch deserialization perf

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
The batch request deserialization is CPU intensive (especially at scale), however it's parallelizable.

This makes it about 5x faster in preprod test.

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ian-shim — APPROVED** [review]
🚀

File: node/grpc/utils.go [hunk 1]
```diff
@@ -13,11 +13,16 @@ import (
...
 	"github.com/consensys/gnark-crypto/ecc/bn254"
 	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
+	"github.com/gammazero/workerpool"
 	"github.com/wealdtech/go-merkletree"
 	"github.com/wealdtech/go-merkletree/keccak256"
 	"google.golang.org/protobuf/proto"
 )
 
+var (
+	maxDeserializationWorkers = 32
```
Comment (pschork) [inline node/grpc/utils.go:23]: would be nice to expose this in operator config for tuning purposes

**Review by dmanc — APPROVED** [review]

**Review by pschork — APPROVED** [review]
🙏

File: node/grpc/utils.go (hunk 1 above)
Comment (jianoaix) [reply node/grpc/utils.go]: sounds good, I didn't make it so as I was thinking the validator was not using flag yet and wanted to do it together separately later (validation is actually using flag)

**Pull request merged**